src/utilities/createResolvedRoute.ts

```diff
@@ -0,0 +1,34 @@
+import { markRaw } from 'vue'
+import { Route, Param, Resolved, routeDepthSymbol, isRejectionSymbol } from '@/types'
+
+type ResolvedRouteProperties<T extends Route> = {
+  matched: T,
+  matches: Route[],
+  name: string,
+  path: string,
+  query: string,
+  params: Record<string, Param[]>,
+  depth: number,
+  isRejection?: true,
+}
```

Comment:
Many of these properties are the same as `Resolved<Route>` right? Would it be better to define this as resolved route & the extra properties used to assign to symbols?

Reply:
I tried that originally but the omit and add type was kinda hard to read. This seemed simpler even if it's less dry